### ISSUE TITLE
about DANN`s code

### ISSUE THREAD

**Issue by fanucai** [Feaxure-fresh/TL-Bearing-Fault-Diagnosis#9]
Thank you very much for your excellent work.
I have a little question about DANN's code. The main point of the original paper seems to emphasize the necessity of maximizing loss_d to obtain the optimal loss and setting lambda (i.e., the tradeoff) to a negative value. However, it appears that this requirement is not reflected in the code, as the tradeoff remains positive and there is no explicit negation of the tradeoff in the get_tradeoff function.  I am wondering is this a mistake, or was it intentionally designed this way? 
I apologize for the minor detail, but could you please confirm it that's been bothering me? I will appreciate it.

**Comment by Feaxure-fresh** [comment]
Thank you for your question regarding the implementation of the Domain-Adversarial Neural Network (DANN). In our implementation, we actually make use of the Gradient Reversal Layer (with positive tradeoff parameter) rather than the negative tradeoff parameter to implement the min-max game of DANN. During backpropagation, the GRL reverses the gradient direction, which aligns with our goal of maximizing the domain classification loss (loss_d) through the feature extractor. This is why lambda is set to a positive value, namely, the GRL automatically takes care of the necessary inversion.

I hope this clarifies the rationale behind the implementation choices.

**Comment by fanucai** [comment]
Thank you for your prompt confirmation and reply. I understand the gist, but I will need to review the details. I may have further questions later on, would you please hold this issue for a while? Anyway, I greatly appreciate your cooperation.

**Comment by fanucai** [comment]
Thank you for waiting. As you mentioned, I found the Gradient Reversal Layer automatically takes care of the necessary inversion and I have cleared my doubts. Thank you again sincerely.
